equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java

```diff
@@ -263,12 +264,28 @@ public Equipment update(@RequestBody Equipment equipment){
         }
         
         validateChannelNum(equipment);
+        
+        Equipment existingEquipment = equipmentDatastore.getOrNull(equipment.getId());
+        ApElementConfiguration existingApElementConfig = null;
+        if (existingEquipment != null && existingEquipment.getDetails() instanceof ApElementConfiguration) {
+            existingApElementConfig = (ApElementConfiguration) existingEquipment.getDetails();
+        }
 
         Equipment ret = equipmentDatastore.update(equipment);
+        ApElementConfiguration updatedApElementConfig = null;
+        if (ret != null && ret.getDetails() instanceof ApElementConfiguration) {
+            updatedApElementConfig = (ApElementConfiguration) ret.getDetails();
+        }
 
         LOG.debug("Updated Equipment {}", ret);
-      
-        EquipmentChangedEvent event = new EquipmentChangedEvent(ret);
+        
+        EquipmentChangedEvent event;
+        if (existingApElementConfig != null && updatedApElementConfig != null &&
+                updatedApElementConfig.needsToBeUpdatedOnDevice(existingApElementConfig)) {
+            event = new EquipmentApImpactingChangedEvent(ret);
+        } else {
+            event = new EquipmentChangedEvent(ret);
```

Comment:
I think we should always send EquipmentChangedEvent to indicate that the equipment record changed. Similar to the handling for LocationChangedApImpactingEvent. 

Naming would be more consistent if this was EquipmentChangedApImpactingEvent (to match existing LocationChangedApImpactingEvent)

Reply:
EquipmentChangedApImpactingEvent is extended from EquipmentChangedEvent. All three events, EquipmentApImpactingChangedEvent, EquipmentChannelsChangedEvent and EquipmentCellSizeAttributesChangedEvent, update equipment record. 
I am thinking other way around, to rewrite LocationChangedApImpactingEvent, extended from LocationChangedEvent instead of creating a new CustomerEventWithPayload, so there will be no more two events created.

Reply:
There's lots of stream processors which are looking for LocationChangedEvent and don't care about the other details. The two events were by intention at the time